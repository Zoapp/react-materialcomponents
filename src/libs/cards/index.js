/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import CardActions from "./actions";
import Card from "./card";
import CardMedia from "./media";
import CardText from "./text";
import CardPrimary from "./primary";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";

export {
  Card,
  CardActions,
  CardMedia,
  CardText,
  CardPrimary,
  CardTitle,
  CardSubtitle,
};
