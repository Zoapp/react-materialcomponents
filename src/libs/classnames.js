/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
  Rewritten to ES7 by Mik BRY
*/
const ClassNames = (args) => {
  const classes = [];
  args.forEach((arg) => {
    if (arg) {
      const argType = typeof arg;
      if (argType === "string" || argType === "number") {
        classes.push(arg);
      } else if (Array.isArray(arg) && arg.length) {
        const inner = ClassNames(arg);
        if (inner) {
          classes.push(inner);
        }
      } else if (argType === "object") {
        Object.keys(arg).forEach((key) => {
          if (arg[key]) {
            classes.push(key);
          }
        });
      }
    }
  });
  return classes.join(" ");
};

export default ClassNames;
