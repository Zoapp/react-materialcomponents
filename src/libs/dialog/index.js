/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Dialog from "./dialog";
import DialogBody from "./body";
import DialogFooter from "./footer";
import DialogHeader from "./header";

export { Dialog, DialogBody, DialogFooter, DialogHeader };
