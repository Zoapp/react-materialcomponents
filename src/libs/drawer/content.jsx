/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import PropTypes from "prop-types";
import Rmdc from "../";

/**
 * mdc-drawer__content
 * See
 * https://material.io/components/web/catalog/drawer/
 *
 */

const MDC_DRAWERCONTENT = "mdc-drawer__content";

const DrawerContent = ({
  children, list, ...props
}) => {
  let classes = MDC_DRAWERCONTENT;
  if (list) {
    classes += " mdc-list";
  }
  return Rmdc.render(<nav className={classes} >{children}</nav>, props);
};

DrawerContent.defaultProps = {
  mdcElement: MDC_DRAWERCONTENT,
  children: null,
  list: false,
};

DrawerContent.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
  list: PropTypes.bool,
};

export default DrawerContent;
