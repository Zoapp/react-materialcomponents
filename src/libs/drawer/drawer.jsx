/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import Rmdc from "../";

/**
 * See:
 * https://material.io/components/web/catalog/drawers/
 * https://material-components-web.appspot.com/drawer/persistent-drawer.html
 *
 * TODO:
 * - temporary, permanent, persistent
 */

const MDC_DRAWER = "mdc-drawer";

export default class Drawer extends PureComponent {
  render() {
    const {
      children, type, ...props
    } = this.props;
    const classes = `${MDC_DRAWER} mdc-drawer--${type} mdc-typography`;

    if (type === "permanent") {
      return Rmdc.render(<nav className={classes}>{children}</nav>, props);
    }
    return Rmdc.render(
      (
        <aside className={classes}>
          <nav className="mdc-drawer__drawer">{children}</nav>
        </aside>),
      props,
    );
  }
}

Drawer.defaultProps = {
  mdcElement: MDC_DRAWER,
  children: null,
  type: "permanent",
};

Drawer.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
  type: PropTypes.string,
};
