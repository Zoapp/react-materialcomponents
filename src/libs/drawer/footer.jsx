/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import PropTypes from "prop-types";

const MDC_DRAWERFOOTER = "mdc-drawer__footer";

const DrawerFooter = ({ children }) => {
  const classes = MDC_DRAWERFOOTER;

  return (
    <footer className={classes}>
      <div className={`${classes}-content`}>{children}</div>
    </footer>
  );
};

DrawerFooter.propTypes = {
  children: PropTypes.node,
};

export default DrawerFooter;
