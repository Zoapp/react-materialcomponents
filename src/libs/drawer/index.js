import Drawer from "./drawer";
import DrawerContent from "./content";
import DrawerHeader from "./header";

export { Drawer, DrawerContent, DrawerHeader };
