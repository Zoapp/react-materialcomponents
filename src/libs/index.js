/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Rmdc from "./rmdc";
import Button from "./components/button";
import Icon from "./components/icon";
import Content from "./components/content";
import Checkbox from "./components/checkbox";
import Fab from "./components/fab";
import TextField from "./components/textfield";
import Radio from "./components/radio";
import FormField from "./components/formField";
import Select from "./components/select";
import { Menu, MenuAnchor, MenuItem, MenuDivider } from "./menu";
import Snackbar from "./components/snackbar";
import LinearProgress from "./components/linearProgress";
import Switch from "./components/switch";
import Slider from "./components/slider";
import IconToggle from "./components/iconToggle";
import { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon } from "./toolbar";
import { Drawer, DrawerContent, DrawerHeader } from "./drawer";
import { List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader } from "./list";
import { Grid, Inner, Cell } from "./layoutGrid";
import { Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle } from "./cards";
import { Tabbar, Tab } from "./tab";
import { GridList, Tile } from "./gridList";
import { Dialog, DialogHeader, DialogBody, DialogFooter } from "./dialog";

export {
  Button, Content, Fab, TextField, Checkbox, Radio, FormField, Select, Snackbar,
  LinearProgress, Switch, Slider, IconToggle, Icon,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerContent, DrawerHeader,
  List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader,
  Grid, Inner, Cell,
  Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle,
  Tabbar, Tab,
  GridList, Tile,
  Dialog, DialogHeader, DialogBody, DialogFooter,
  Menu, MenuAnchor, MenuItem, MenuDivider,
};

export default Rmdc;
