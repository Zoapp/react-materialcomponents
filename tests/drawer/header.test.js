import React from "react";
import renderer from "react-test-renderer";
import { DrawerHeader } from "libs/drawer";

describe("drawer/DrawerHeader", () => {
  it("can have children", () => {
    const tree = renderer
      .create(<DrawerHeader>Children is Here!</DrawerHeader>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a listed prop", () => {
    const tree = renderer.create(<DrawerHeader listed="permanent" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
