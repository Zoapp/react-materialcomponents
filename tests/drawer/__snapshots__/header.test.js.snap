// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawer/DrawerHeader can have a listed prop 1`] = `
<header
  className="mdc-drawer__header"
>
  <div
    className="mdc-drawer__header-content"
  />
</header>
`;

exports[`drawer/DrawerHeader can have children 1`] = `
<header
  className="mdc-drawer__header"
>
  <div
    className="mdc-drawer__header-content"
  >
    Children is Here!
  </div>
</header>
`;
